public/app/features/datasources/settings/DataSourceSettingsPage.tsx

```diff
@@ -95,6 +98,19 @@ export class DataSourceSettingsPage extends PureComponent<Props, State> {
     }
   }
 
+  importPlugin = (cancelablePromise: CancelablePromise<any>) => {
+    this.importPluginPromise = cancelablePromise;
+    this.importPluginPromise.promise
+      .then((importedPlugin: GenericDataSourcePlugin) => {
+        this.setState({ plugin: importedPlugin });
+      })
+      .catch(({ isCanceled }) => {
```

Comment:
This `catch` if never going to fire as the promise is never cancelled. You need to make sure it's cancelled when component unmounts

Reply:
I dropped the network in devtools to 3G and stopped it from loading to get the error about setting state after unmount. After adding this, the error no longer happens so I do think it is being canceled.

Reply:
I think that may be because this is also catching the error thrown by `this.setState` on line before? In general I would also expect the promise to be cancelled in `componentWillUnmount` if want to use cancelation as a pattern here.